devicetypes/smartthings/smartsense-moisture-sensor.src/smartsense-moisture-sensor.groovy

```diff
@@ -255,52 +255,54 @@ private Map getMoistureResult(value) {
 	]
 }
 
-def refresh()
-{
+def refresh() {
 	log.debug "Refreshing Temperature and Battery"
-	[
-		
-
+	def refreshCmds = [
         "st rattr 0x${device.deviceNetworkId} 1 0x402 0", "delay 200",
-		"st rattr 0x${device.deviceNetworkId} 1 1 0x20"
-
+		"st rattr 0x${device.deviceNetworkId} 1 1 0x20", "delay 200"
 	]
+
+	return refreshCmds + enrollResponse()
 }
 
 def configure() {
-
-	String zigbeeId = swapEndianHex(device.hub.zigbeeId)
-	log.debug "Confuguring Reporting, IAS CIE, and Bindings."
+	String zigbeeEui = swapEndianHex(device.hub.zigbeeEui)
+	log.debug "Configuring Reporting, IAS CIE, and Bindings."
 	def configCmds = [	
-		"zcl global write 0x500 0x10 0xf0 {${zigbeeId}}", "delay 200",
-		"send 0x${device.deviceNetworkId} 1 1", "delay 1500",
+		"zcl global write 0x500 0x10 0xf0 {${zigbeeEui}}", "delay 200",
+		"send 0x${device.deviceNetworkId} 1 1", "delay 500",
         
         "zcl global send-me-a-report 1 0x20 0x20 300 0600 {01}", "delay 200",
-        "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
+        "send 0x${device.deviceNetworkId} 1 1", "delay 500",
         
         "zcl global send-me-a-report 0x402 0 0x29 300 3600 {6400}", "delay 200",
-        "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
+        "send 0x${device.deviceNetworkId} 1 1", "delay 500",
         
         
 		"zdo bind 0x${device.deviceNetworkId} 1 1 0x402 {${device.zigbeeId}} {}", "delay 500",
-		"zdo bind 0x${device.deviceNetworkId} 1 1 0x001 {${device.zigbeeId}} {}", "delay 1000",
-        
-        "raw 0x500 {01 23 00 00 00}", "delay 200",
-        "send 0x${device.deviceNetworkId} 1 1", "delay 1000",
+		"zdo bind 0x${device.deviceNetworkId} 1 1 0x001 {${device.zigbeeId}} {}", "delay 500"
 	]
     return configCmds + refresh() // send refresh cmds as part of config
 }
 
 def enrollResponse() {
 	log.debug "Sending enroll response"
-    [	
-    	
-	"raw 0x500 {01 23 00 00 00}", "delay 200",
-    "send 0x${device.deviceNetworkId} 1 1"
-        
+	String zigbeeEui = swapEndianHex(device.hub.zigbeeEui)
+    [
+		//Resending the CIE in case the enroll request is sent before CIE is written
+		"zdo bind 0x${device.deviceNetworkId} 1 ${endpointId} 1 {${device.zigbeeId}} {}", "delay 200",
+		"zcl global write 0x500 0x10 0xf0 {${zigbeeEui}}",
+		"send 0x${device.deviceNetworkId} 1 ${endpointId}", "delay 500",
+		//Enroll Response
+		"raw 0x500 {01 23 00 00 00}",
+		"send 0x${device.deviceNetworkId} 1 1", "delay 200"
     ]
 }
 
+private getEndpointId() {
```

Comment:
See the calls to endpointId:  Groovy "helpfully" converts these calls to include a camel-cased "get" on the front.